cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -337,30 +337,93 @@ std::pair<VersionedItem, FrameAndDescriptor> LocalVersionedEngine::read_datafram
     return std::make_pair(version.value_or(VersionedItem{}), std::move(frame_and_descriptor));
 }
 
-std::pair<VersionedItem, std::optional<google::protobuf::Any>> LocalVersionedEngine::read_descriptor_version_internal(
-        const StreamId& stream_id,
-        const VersionQuery& version_query
+folly::Future<std::pair<DescriptorItem, std::optional<google::protobuf::Any>>> LocalVersionedEngine::get_descriptor(
+    AtomKey&& key){
+    return store()->read(key)
+    .thenValue([](auto&& key_seg_pair) -> std::pair<DescriptorItem, std::optional<google::protobuf::Any>>{
+        auto key_seg = std::move(std::get<0>(key_seg_pair));
+        auto seg = std::move(std::get<1>(key_seg_pair));
+        auto meta = seg.has_metadata() ? std::make_optional<google::protobuf::Any>(*seg.metadata()) : std::nullopt;
+        auto start_index = seg.column(position_t(index::Fields::start_index)).type().visit_tag([&](auto column_desc_tag) -> NumericIndex {
+            using ColumnDescriptorType = std::decay_t<decltype(column_desc_tag)>;
+            using ColumnTagType =  typename ColumnDescriptorType::DataTypeTag;
+            if constexpr (is_numeric_type(ColumnTagType::data_type)) {
+                NumericIndex start_index = std::numeric_limits<NumericIndex>::max();
+                auto column_data = seg.column(position_t(index::Fields::start_index)).data();
+                while (auto block = column_data.template next<ColumnDescriptorType>()) {
+                    auto ptr = const_cast<NumericIndex *>(reinterpret_cast<const NumericIndex *>(block.value().data()));
+                    const auto row_count = block.value().row_count();
+                    for (auto i = 0u; i < row_count; ++i) {
+                        auto value = static_cast<NumericIndex>(*ptr++);
+                        start_index = value < start_index ? value : start_index;
+                    }
+                }
+                return start_index;
+            }else {
+                util::raise_rte("Unsoprted index type {}", seg.column(position_t(index::Fields::start_index)).type());
+            }
+        });
+
+        auto end_index = seg.column(position_t(index::Fields::end_index)).type().visit_tag([&](auto column_desc_tag) -> NumericIndex {
+            using ColumnDescriptorType = std::decay_t<decltype(column_desc_tag)>;
+            using ColumnTagType =  typename ColumnDescriptorType::DataTypeTag;
+            if constexpr (is_numeric_type(ColumnTagType::data_type)) {
+                NumericIndex end_index = 0;
```

Comment:
Timestamps are signed! Use `std::numeric_limits<NumericIndex>::lowest`

Reply:
we will use optional as discussed